plugins/database/cassandra/cassandra.go

```diff
@@ -80,162 +71,174 @@ func (c *Cassandra) getConnection(ctx context.Context) (*gocql.Session, error) {
+	if req.Password != nil {
+		err := c.changeUserPassword(ctx, req.Username, req.Password)
+		return newdbplugin.UpdateUserResponse{}, err
+	}
+	// Expiration is no-op
+	return newdbplugin.UpdateUserResponse{}, nil
+}
 
+func (c *Cassandra) changeUserPassword(ctx context.Context, username string, changePass *newdbplugin.ChangePassword) error {
 	session, err := c.getConnection(ctx)
 	if err != nil {
 		return err
 	}
 
-	revocationCQL := statements.Revocation
-	if len(revocationCQL) == 0 {
-		revocationCQL = []string{defaultUserDeletionCQL}
+	rotateCQL := changePass.Statements.Commands
+	if len(rotateCQL) == 0 {
+		rotateCQL = []string{defaultChangePasswordCQL}
 	}
 
 	var result *multierror.Error
-	for _, stmt := range revocationCQL {
+	for _, stmt := range rotateCQL {
 		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
 			query = strings.TrimSpace(query)
 			if len(query) == 0 {
 				continue
 			}
 
-			err := session.Query(dbutil.QueryHelper(query, map[string]string{
+			m := map[string]string{
 				"username": username,
-			})).WithContext(ctx).Exec()
-
+				"password": changePass.NewPassword,
+			}
+			err := session.
+				Query(dbutil.QueryHelper(query, m)).
+				WithContext(ctx).
+				Exec()
 			result = multierror.Append(result, err)
 		}
 	}
 
 	return result.ErrorOrNil()
 }
 
-func (c *Cassandra) RotateRootCredentials(ctx context.Context, statements []string) (map[string]interface{}, error) {
-	// Grab the lock
+// DeleteUser attempts to drop the specified user.
+func (c *Cassandra) DeleteUser(ctx context.Context, req newdbplugin.DeleteUserRequest) (newdbplugin.DeleteUserResponse, error) {
 	c.Lock()
 	defer c.Unlock()
 
 	session, err := c.getConnection(ctx)
 	if err != nil {
-		return nil, err
-	}
-
-	rotateCQL := statements
-	if len(rotateCQL) == 0 {
-		rotateCQL = []string{defaultRootCredentialRotationCQL}
+		return newdbplugin.DeleteUserResponse{}, err
 	}
 
-	password, err := c.GeneratePassword()
-	if err != nil {
-		return nil, err
+	revocationCQL := req.Statements.Commands
+	if len(revocationCQL) == 0 {
+		revocationCQL = []string{defaultUserDeletionCQL}
 	}
 
 	var result *multierror.Error
-	for _, stmt := range rotateCQL {
+	for _, stmt := range revocationCQL {
 		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
 			query = strings.TrimSpace(query)
 			if len(query) == 0 {
 				continue
 			}
 
-			err := session.Query(dbutil.QueryHelper(query, map[string]string{
-				"username": c.Username,
-				"password": password,
-			})).WithContext(ctx).Exec()
+			m := map[string]string{
+				"username": req.Username,
+			}
+			err := session.
+				Query(dbutil.QueryHelper(query, m)).
+				WithContext(ctx).
+				Exec()
 
 			result = multierror.Append(result, err)
 		}
 	}
 
-	err = result.ErrorOrNil()
-	if err != nil {
-		return nil, err
-	}
-
-	c.rawConfig["password"] = password
-	return c.rawConfig, nil
+	return newdbplugin.DeleteUserResponse{}, result.ErrorOrNil()
```

Comment:
Meant to ask this offline earlier, but can you remind me why the interface doesn't return pointer response values so that we can return `nil`?

Reply:
There isn't a particular need to do so. Plus it opens up the problem of having to do nil checks in other parts of the code and having some potentially inconsistent or confusing behavior.